### ISSUE TITLE
Test cannot be run against typescript@next

### ISSUE THREAD

**Issue by sledorze** [gcanti/monocle-ts#80]
I got this error:
```
Error: /Users/sledorze/projects/monocle-ts/dtslint/index.ts:71:85
ERROR: 71:85  expect  TypeScript@next compile error:
Argument of type '["a", "b", "c", "d"]' is not assignable to parameter of type '[keyof T]'.
  Types of property 'length' are incompatible.
    Type '4' is not assignable to type '1'.
```

**Comment by gcanti** [comment]
@sledorze do you happen to know whether there is a way in dtslint to not run type level tests against `typescript@next`? 
I can't find it in the documentation.

I see three options:

- change / comment the offending test
- find a way to not run type level tests against `typescript@next`
- in `io-ts` I switched to a customized fork of dtslint, we could do the same here

**Comment by gcanti** [comment]
> in `io-ts` I switched to a customized fork of dtslint, we could do the same here

@sledorze as a temporary fix I'll go with this (by merging the `1.6.0` branch, we can try to restore the official dtslint later)

**Comment by gcanti** [comment]
Closing via https://github.com/gcanti/monocle-ts/pull/81
